tools/discovery/discovery.go

```diff
@@ -0,0 +1,131 @@
+package discovery
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+
+	"github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+	"github.com/xiekeyang/oci-discovery/tools/object"
+)
+
+var (
+	defaultTrans = &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
I'd rather not have this as part of any landed code.  Testing should [mock the network access][1], and folks who do not have a valid X.509 cert for secure HTTPS should just be using HTTP (instead of having clients ignore their broken cert).

[1]: https://github.com/xiekeyang/oci-discovery/blob/44ec3cf3113e29a743ad04220ccb7ff5197dab2a/oci_discovery/ref_engine_discovery/test.py#L54-L56